git/foundations_git/commit_messages.md

```diff
@@ -20,14 +20,47 @@ Yes! Let me give you a quick list of reasons why:
 
 -   Having a good commit message history will also be helpful if you come back to a project you were working on after stepping away from it for a while. You likely won't remember your thought process and changes made when initially writing out the code. 
 
+### Bad vs. Good Commits
+
+When it comes to writing commits, it is crucial to know how to write them effectively. Here's an example of a bad commit message:
+
+`fix a bug`
+
+Even though it describes what you did, the message is too vague, which leaves the other developers on your team confused. A good commit message will explain the why behind your changes. In other words, a commit message describes what problem your changes solve and how it solves them. 
+
+Effective commits consist of two separate parts: a subject, and a body:
+
+## Subject
+A brief summary of the change you made to the project.  **Note:** GitHub has a 50-character limit so we recommend keeping your commits' subject to within this amount. 
+
+`This is the change I made to the codebase.`
+
+## Body
+A concise yet clear description of what you did.  
+
+`Describe the problem that your changes solve and how in 50 words or less`
+
+
+Now that we learned the secret to creating a good commit message, let's try and fix the commit message from earlier:
+
+`Add missing link and alt text to company's logo`
+`Screen readers won't read the images to users with disabilities without this information`
+
+Ahh, that's better! :) Now, developers can gain a better understanding of this commit message because it does the following:
+
+* Provides a subject that specifies your code's action (e.g., "Add missing link and alt text company's logo")
+* Contains a body that provides a concise yet clear description of what is created (e.g., "Screen readers won't read the images to users with disabilities without this information")
```

Comment:
Nit: Just minor rewording to drive home good commit messages primarily provide the "why" for future readers:

```suggestion
* Contains a body that provides a concise yet clear description of why the commit needed to be made (e.g., "Screen readers won't read the images to users with disabilities without this information")
```